pkg/engine/shield/add.go

```diff
@@ -4,20 +4,46 @@ import (
 	"github.com/simimpact/srsim/pkg/engine/event"
 	"github.com/simimpact/srsim/pkg/engine/info"
 	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
 )
 
 func (mgr *Manager) AddShield(id key.Shield, shield info.Shield) {
 	// TODO: LOGIC FOR ADDING A SHIELD
 	// 1. Check if the target already has this shield, if so remove old (mgr.RemoveShield)
+	if mgr.HasShield(shield.Target, id) {
+		mgr.RemoveShield(id, shield.Target)
+	}
```

Comment:
nit: this is inefficient, you are looping over the shields list twice (once to find if you have the shield, 2nd to remove in the worst case). Better if we can reduce this to one loop.

Also I am not sure if the right approach is to emit the `RemoveShield` event or not. Based on the latest TC findings, we **do not** emit a `Remove` event in the case of modifiers being replaced, so it is probably best to do the same here